packages/engineer/feature/dev-server.dev-server.env.ts

```diff
@@ -0,0 +1,282 @@
+import devServerFeature, { devServerEnv } from './dev-server.feature';
+import {
+    launchHttpServer,
+    NodeEnvironmentsManager,
+    IRunFeatureOptions,
+    IFeatureMessagePayload,
+} from '@wixc3/engine-scripts/src/';
+import { ApplicationProxyService } from '../src/application-proxy-service';
+import express from 'express';
+import {
+    ensureTopLevelConfigMiddleware,
+    createCommunicationMiddleware,
+    createLiveConfigsMiddleware,
+    createConfigMiddleware,
+    OverrideConfig,
+} from '@wixc3/engine-scripts/src/config-middleware';
+import WebpackDevMiddleware from 'webpack-dev-middleware';
+import { createFeaturesEngineRouter } from '@wixc3/engine-scripts/src/engine-router';
+import webpack from 'webpack';
+
+import { WsServerHost } from '@wixc3/engine-core-node';
+
+import { generateConfigName } from '@wixc3/engine-scripts/src/engine-router';
+import performance from '@wixc3/cross-performance';
+
+devServerFeature.setup(
+    devServerEnv,
+    (
+        {
+            run,
+            devServerConfig: {
+                httpServerPort,
+                featureName,
+                singleFeature,
+                publicConfigsRoute,
+                title,
+                publicPath,
+                configName,
+                singleRun,
+                inspect,
+                autoLaunch,
+                nodeEnvironmentsMode,
+                basePath,
+                mode,
+                overrideConfig,
+                defaultRuntimeOptions,
+            },
+            engineerWebpackConfigs,
+            serverListeningHandlerSlot,
+        },
+        { COM: { communication } }
+    ) => {
+        const application = new ApplicationProxyService({ basePath });
+        const overrideConfigsMap = new Map<string, OverrideConfig>();
+        const disposables = new Set<() => unknown>();
+        let nodeEnvManager: NodeEnvironmentsManager | null = null;
+
+        const getNodeEnvManager = () => nodeEnvManager;
+
+        // Extract these into a service
+        const runFeature = async ({
```

Comment:
The next 4 methods (up to close) should probably move into application-proxy, which would eventually be the main interface of managing the app